tokens/src/thin_client.rs

```diff
@@ -0,0 +1,174 @@
+use solana_client::rpc_client::RpcClient;
+use solana_runtime::bank_client::BankClient;
+use solana_sdk::{
+    account::Account,
+    client::{AsyncClient, SyncClient},
+    fee_calculator::FeeCalculator,
+    hash::Hash,
+    message::Message,
+    pubkey::Pubkey,
+    signature::{Signature, Signer},
+    signers::Signers,
+    system_instruction,
+    sysvar::{
+        recent_blockhashes::{self, RecentBlockhashes},
+        Sysvar,
+    },
+    transaction::Transaction,
+    transport::{Result, TransportError},
+};
+use solana_transaction_status::TransactionStatus;
+
+pub trait Client {
```

Comment:
After this is merged, first course of action will be looking for ways to get rid of trait.